### PR TITLE
Fix: #5186: MissingDefaultResource + AppCompatCustomView + FragmentTagUsage

### LINKED ISSUE #5186: MissingDefaultResource + AppCompatCustomView + FragmentTagUsage
## Description 
 We would like to fix all the remaining lint issues on Oppia Android in the "MissingDefaultResource + AppCompatCustomView + FragmentTagUsage" category.

## Steps To Reproduce
Run the following command from a terminal in a checked-out Oppia Android repo:

`./gradlew :app:lint`

**Output:** 

- MissingDefaultResource mainly has a single issue with a string which has different names in english and other translations. It can be easily fixed, by matching the name of that string in english and other translations.
- For AppCompatCustomView, it can be fixed by replacing the Normal views with AppCompat views
- <fragment/> tags can be replaced with <androidx.fragment.app.FragmentContainerView/>.

## Expected Behavior
Running the command above should not show any errors.

### PR DESCRIPTION
**Pull request by theMr17**
## Explanation

Fixes #5186 [part of #5169]

1. `MissingDefaultResource` has been fixed already by https://github.com/oppia/oppia-android/pull/5172. It had issues with these plurals:
   -  [minutes_ago](https://github.com/oppia/oppia-android/blob/e060e2ff265ce35c92f1b969f6d7ad002d1a62ad/app/src/main/res/values/strings.xml#L456)
   - [hours_ago](https://github.com/oppia/oppia-android/blob/e060e2ff265ce35c92f1b969f6d7ad002d1a62ad/app/src/main/res/values/strings.xml#L461C18-L461C27)
   -  [days_ago](https://github.com/oppia/oppia-android/blob/e060e2ff265ce35c92f1b969f6d7ad002d1a62ad/app/src/main/res/values/strings.xml#L465C18-L465C26)

3. For `AppCompatCustomView`, it is fixed by replacing the Normal views with AppCompat views

4. `<fragment/>` tags are replaced with `<androidx.fragment.app.FragmentContainerView/>`, with some changes in the NavigationDrawerFragment to ensure that binding has been initialized before it is accessed.

## Screen Recording
https://github.com/oppia/oppia-android/assets/84731134/3a340cff-70a4-4e77-a1a3-e5b7a8c71a6e

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).
<br>

### CODE REVIEW

File: app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragment.kt [hunk 1]
```diff
@@ -30,11 +34,16 @@ class NavigationDrawerFragment :
...
+      navigationDrawerFragmentPresenter.handleCreateView(inflater, container)
+    navigationDrawerFragmentPresenter.setUpDrawer(drawerLayout, toolbar, menuItemId)
+    return navigationDrawerFragmentView
   }
 
   fun setUpDrawer(drawerLayout: DrawerLayout, toolbar: Toolbar, menuItemId: Int) {
-    navigationDrawerFragmentPresenter.setUpDrawer(drawerLayout, toolbar, menuItemId)
+    this.drawerLayout = drawerLayout
+    this.toolbar = toolbar
+    this.menuItemId = menuItemId
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragment.kt]: Could you please move this to the `navigationDrawerFragmentPresenter`, and please use databinding---initialize bindings for this particular layout and use it.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @theMr17!

Please add a video demonstrating that the navigation drawer behaves as expected with the initialization changes, for all the menu options.

**Comment by theMr17** [comment]
@adhiamboperes PTAL

**Review by adhiamboperes — APPROVED** [review]
Thanks @theMr17, this LGTM!

**Pull request merged**